fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/CreateResourceServiceImpl.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.impl.services;
+
+import static org.fcrepo.kernel.api.FedoraTypes.FEDORA_PAIRTREE;
+import static org.fcrepo.kernel.api.RdfLexicon.NON_RDF_SOURCE;
+import static org.fcrepo.kernel.impl.services.functions.resourceServiceFunctions.checkAclLinkHeader;
+import static org.fcrepo.kernel.impl.services.functions.resourceServiceFunctions.determineInteractionModel;
+import static org.fcrepo.kernel.impl.services.functions.resourceServiceFunctions.hasRestrictedPath;
+
+import org.fcrepo.kernel.api.ExternalContent;
+import org.fcrepo.kernel.api.exception.CannotCreateResourceException;
+import org.fcrepo.kernel.api.exception.ItemNotFoundException;
+import org.fcrepo.kernel.api.exception.RepositoryRuntimeException;
+import org.fcrepo.kernel.api.models.ResourceHeaders;
+import org.fcrepo.kernel.api.operations.RdfSourceOperationFactory;
+import org.fcrepo.kernel.api.operations.ResourceOperation;
+import org.fcrepo.kernel.api.services.CreateResourceService;
+import org.fcrepo.kernel.api.services.functions.UniqueValueSupplier;
+import org.fcrepo.persistence.api.PersistentStorageSession;
+import org.fcrepo.persistence.api.PersistentStorageSessionManager;
+import org.fcrepo.persistence.api.exceptions.PersistentItemNotFoundException;
+import org.fcrepo.persistence.api.exceptions.PersistentStorageException;
+
+import javax.inject.Inject;
+import javax.ws.rs.core.Link;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CreateResourceServiceImpl implements CreateResourceService {
```

Comment:
this is a very HTTP POST oriented approach, which is probably fine. If we ever have a need for creating resources outside of the http layer then we would probably need smaller services. It would be good to have a consistent approach across services as to whether or RDF and non-RDF sources happen in the same service.

Reply:
This is what I meant by knowing the operation, if I pass in POST or PUT then I could make changes. But that would make this much larger. 

It might make more sense to have an actual `CreateResourceServiceWrapper` that takes the HTTP method and all the arguments and passes off to `CreateResourceServiceImpl` or `CreateOrReplaceResourceServiceImpl`

Reply:
Or I guess just calling the appropriate method in `FedoraLdp` makes more sense 🤦‍♂